src/main/java/com/_up/megastore/controllers/interfaces/IProductController.java

```diff
@@ -22,7 +22,18 @@ ProductResponse saveProduct(
             @RequestPart MultipartFile multipartFile
     );
 
-    @DeleteMapping("/{productId}")
-    @ResponseStatus(HttpStatus.NO_CONTENT)
-    void deleteProduct(@PathVariable UUID productId);
+    @PatchMapping(value = "/{productId}", consumes = { MediaType.MULTIPART_FORM_DATA_VALUE, MediaType.APPLICATION_JSON_VALUE })
+    @ResponseStatus(HttpStatus.OK)
+    ProductResponse updateProduct(
+            @PathVariable UUID productId,
+            @RequestPart @Valid UpdateProductRequest updateProductRequest,
+            @RequestPart @Nullable MultipartFile multipartFile
+    );
+
+    @DeleteMapping(value = "/{productId}")
+    @ResponseStatus(HttpStatus.OK)
+    ProductResponse deleteProduct(
+            @PathVariable UUID productId
+    );
```

Comment:
Creo que acá no es correcto devolver el producto puesto que lo estamos 'eliminando'. Dejaría la función como void y en vez de devolver un OK devolvería un NO_CONTENT que significa que la petición se realizó con éxito pero que no hay nada que mostrar
```suggestion
    @ResponseStatus(HttpStatus.NO_CONTENT)
    void deleteProduct(
            @PathVariable UUID productId
    );
```

Reply:
No estoy de acuerdo con esto, por si bien es cierto que lo eliminamos, es un borrado lógico. El front podría utilizar esta actualización para marcar el producto de alguna forma al administrador o algo por el estilo, no es lo mismo que un borrado lógico en el que deja de existir, por lo que no tendría sentido devolverlo.

Esto lo dejo a criterio del equipo cómo quieren manejarlo

Reply:
Me parece que desde el front no es necesaria esta respuesta (hay mecanismos que podemos usar para obtener nuevamente los productos y con mejores prácticas) pero me gustaría ver la opinión de @delgadomatias

Reply:
Para qué obtendrías de nuevo los productos si ya te lo puede devolver la misma petición de la eliminación ya actualizado? Estas haciendo otra petición más que solo le agrega delay

Reply:
Imagino una acción desde el Frontend que te elimine el producto, para ello, ejecutaremos una función reductora que nos actualice el estado de dicho producto. Por lo tanto, la actualizacion en el lado del servidor se haría a través de la petición HTTP y la actualización del estado del lado del cliente justamente con el manejo del estado global.

Por lo tanto, una vez que eliminemos el producto el frontend ya sabe que producto se elimino y si se elimino de manera correcta. No necesita mas información.

Reply:
Banco

Reply:
Es cierto que en este caso es una boludés, pero no es correcto que el frontend tenga lógica para actualizar los datos en su estado. Qué pasaría si eliminar (o cualquier otra cosa) incluyera cambiar más cosas en esa entidad, tendrías que tener lógica en el front de aquellas cosas que se tienen que cambiar, en vez de aprovechar la respuesta del backend que ya hizo los cambios correspondientes.

El estado de debería actualizarse acorde a los datos que devuelve el backend y no tener lógica para manejar el estado. Pero si quieren dejarlo así y manejar el estado de la app sabiendo ya cuál es la entidad que se borra, es otra opción que va a servir